Master branch fails with -DTHREADSAFE=OFF

As of a the latest commits, I see:

```
[ 80%] Building C object src/CMakeFiles/git2internal.dir/sysdir.c.obj
[ 81%] Building C object src/CMakeFiles/git2internal.dir/thread-utils.c.obj
[ 81%] Building C object src/CMakeFiles/git2internal.dir/threadstate.c.obj
[ 82%] Building C object src/CMakeFiles/git2internal.dir/trace.c.obj
[ 82%] Building C object src/CMakeFiles/git2internal.dir/trailer.c.obj
C:/Users/jeroen/Desktop/rtools-packages/mingw-w64-libgit2/src/libgit2-1.1.0/src/threadstate.c: In function 'git_threadstate_global_init':
C:/Users/jeroen/Desktop/rtools-packages/mingw-w64-libgit2/src/libgit2-1.1.0/src/threadstate.c:138:39: error: 'git_tlsdata_global_shutdown' undeclared (first use in this function); did you mean 'git_threadstate_global_shutdown'?
  return git_runtime_shutdown_register(git_tlsdata_global_shutdown);
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                                       git_threadstate_global_shutdown
C:/Users/jeroen/Desktop/rtools-packages/mingw-w64-libgit2/src/libgit2-1.1.0/src/threadstate.c:138:39: note: each undeclared identifier is reported only once for each function it appears in
C:/Users/jeroen/Desktop/rtools-packages/mingw-w64-libgit2/src/libgit2-1.1.0/src/threadstate.c:139:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
At top level:
C:/Users/jeroen/Desktop/rtools-packages/mingw-w64-libgit2/src/libgit2-1.1.0/src/threadstate.c:130:13: warning: 'git_threadstate_global_shutdown' defined but not used [-Wunused-function]
 static void git_threadstate_global_shutdown(void)
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: warning: unrecognized command line option '-Wno-documentation-deprecated-sync'
make[2]: *** [src/CMakeFiles/git2internal.dir/build.make:1838: src/CMakeFiles/git2internal.dir/threadstate.c.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:207: src/CMakeFiles/git2internal.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

```

This is libgit2 master branch, with gcc 8.3.0 on Windows.

--- a/src/threadstate.c
+++ b/src/threadstate.c
@@ -135,7 +135,7 @@ static void git_threadstate_global_shutdown(void)
 
 int git_threadstate_global_init(void)
 {
-	return git_runtime_shutdown_register(git_tlsdata_global_shutdown);
+	return git_runtime_shutdown_register(git_threadstate_global_shutdown);
 }
 
 git_threadstate *git_threadstate_get(void)

